core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -622,7 +631,8 @@ class LogSegment private[log] (val log: FileRecords,
       () => delete(log.deleteIfExists _, "log", log.file, logIfMissing = true),
       () => delete(offsetIndex.deleteIfExists _, "offset index", lazyOffsetIndex.file, logIfMissing = true),
       () => delete(timeIndex.deleteIfExists _, "time index", lazyTimeIndex.file, logIfMissing = true),
-      () => delete(txnIndex.deleteIfExists _, "transaction index", txnIndex.file, logIfMissing = false)
+      () => delete(txnIndex.deleteIfExists _, "transaction index", txnIndex.file, logIfMissing = false),
+      () => delete(() =>  Files.deleteIfExists(producerStateSnapshot.toPath), "producer state snapshot", producerStateSnapshot, logIfMissing = false)
```

Comment:
For consistency, should we cover producerStateSnapshot in lastModified_= ?

Reply:
Yes thanks!